src/components/AddItemForm.js

```diff
@@ -16,23 +33,61 @@ const AddItemForm = () => {
     setNewItem(e.target.value);
   };
 
+  //Check for duplication:
+  //get the existing items list from firebase
+  //loop through the existing items list to check if there is a match with current item
+  //set itemExist to true if duplication and return it
+
+  const checkDuplication = (newItem) => {
+    let existingData = datas;
```

Comment:
is there a reason why you creating a copy of `instances`? we can loop through the `datas` directly, I've tried in my local and it works just fine. Correct me if I'm wrong please.

Reply:
The only reason why a copy of `datas` is created is to make the code more readable. It makes more sense if I know what the data is. Nevertheless, in this case, the name is not that desciptive and it is kind of obvious to what the data makes reference. I suggest two approaches: either changing the name from existingData to existingItemsData or looping directly through `datas `. What would it be more efficient in your opinion @redapy?

Reply:
we can use a descriptive name from the first. We can use `items` or `existingItems` instead of `datas`.
I prefer just `items`. Since we are fetching it that means it exists.

Reply:
Great suggestion, thanks!